core/src/SYCL/Kokkos_SYCL_Parallel_Reduce.hpp

```diff
@@ -455,9 +455,6 @@ class ParallelReduce<FunctorType, Kokkos::RangePolicy<Traits...>, ReducerType,
                              Kokkos::Experimental::SYCLDeviceUSMSpace>(
           space, m_result_ptr, results_ptr,
           sizeof(*m_result_ptr) * value_count);
-      space.fence(
-          "Kokkos::Impl::ParallelReduce::sycl_direct_launch: fence due to "
-          "inaccessible reducer result location");
```

Comment:
So, is this statement no longer true or is it okay to ignore this because of the final fences at the interface level?

Reply:
The respective code looks like https://github.com/kokkos/kokkos/blob/d19aab9981a2c447e832a7b4eb7b16992328fb14/core/src/Kokkos_Parallel_Reduce.hpp#L1749-L1768. The other overloads are below that one.
In particular, `Impl::ParallelReduceFence` takes care of fencing if necessary. For `parallel_scan` with return value, we currently have an unconditional fence since we don't allow `View`s as return type (#5146 adds that feature).